wire-runtime/src/main/java/com/squareup/wire/TypeAdapter.java

```diff
@@ -28,26 +28,50 @@
 
   final WireType type;
 
-  protected TypeAdapter(WireType type) {
+  TypeAdapter(WireType type) {
     this.type = type;
   }
 
-  /** The serialized size of non-null {@code value}. */
-  public abstract int serializedSize(E value);
+  /**
+   * The size of the non-null data {@code value}. This does not include the size required for
+   * a length-delimited prefix (should the type require one).
```

Comment:
maybe instead of `should the type require one` it's `should the type require one in its current context`. Because the message type requires one, except when it doesn't!